soufi/tests/finders/test_ubuntu_finder.py

```diff
@@ -18,25 +18,13 @@
 
 
 class TestUbuntuFinder(base.TestCase):
-    def tearDown(self):
-        super().tearDown()
-        ubuntu.UbuntuFinder.get_archive.cache_clear()
-
     def make_finder(self, name=None, version=None):
         if name is None:
             name = self.factory.make_string('name')
         if version is None:
             version = self.factory.make_string('version')
         return ubuntu.UbuntuFinder(name, version, SourceType.os)
 
-    def test_init_auths_to_Launchpad_only_once(self):
-        login = self.patch(ubuntu.Launchpad, 'login_anonymously')
-        self.make_finder()
-        self.make_finder()
-        login.assert_called_once_with(
-            "soufi", "production", mock.ANY, version="devel", timeout=30
-        )
-
```

Comment:
Why did you remove this?

Reply:
Because unit tests run with the `dogpile.cache.null` backend, so `login_anonymously` will be called twice.

I could update the test to ensure that in this setup, `login_anonymously` is called twice, but that's not really adding much value.

Reply:
The problem is that it's removed a valid test. Can you set the cache to memory just for that call? It's a *very* slow API call on LP so making sure it's been cached is very valid!

Reply:
The *casus belli* for this entire change is putting caching behavior in the hands of the caller, and the library aggressively not having an opinion about it happening anymore, but OK.  It's easy enough to do.

Reply:
In this case I don't think it should have moved out of lru_cache, but the fact that there's a unified caching infra now made sense. However, it's a blunt tool as it stands as you either get ALL THE THINGS or nothing. This one should generally always get memory cached as it's a low overhead cache but a very slow call.